apps/io-services-cms-webapp/src/watchers/on-service-history-change.ts

```diff
@@ -101,15 +104,28 @@ const toRequestSyncLegacyAction = (
   requestSyncLegacy: cmsToLegacy(serviceHistory),
 });
 
-export const handler: RE.ReaderEither<
-  { item: ServiceHistory },
-  Error,
-  NoAction | RequestSyncLegacyAction
-> = ({ item }) =>
-  pipe(
-    item,
-    O.fromPredicate((itm) => itm.fsm.lastTransition !== SYNC_FROM_LEGACY),
-    O.map(toRequestSyncLegacyAction),
-    O.getOrElse(() => noAction),
-    E.right
-  );
+export const handler =
```

Comment:
why not just:

`ReaderEither<{ item: ServiceHistory, config: IConfig, apimClient: ApiManagementClient}, ...>`

in this way you don't have to pass around the apimClient.

Reply:
Cannot be dono atm, cause we need to edit processBatchOf which is used for all Handler CosmosDbTrigger